Fail to compile with launcher editor version 5.3.1

Hello there!

I'd rather open a discussion than an issue for this, because I'm unsure whether it's a me problem or a code issue -- but discussions aren't open in the repo, sincerely sorry for the "noise".
First things first: thank you so much for making all this work available <3

### **Context**
- I'm using the launcher version of the 5.3.1 editor (_not compiling editor from source_)
- I'm using VSCode
- _I'm already using SUDS within my project (_cloned from repo, compiles fine_)_

**Issue:** As soon as I'm adding the source code of SUQS (_whether cloned or from master .zip_), I can't open my project anymore and get a "Failed to compile, try rebuilding editor from source" error.

> Note: Whether the plugin is registered or not with project.Build.cs doesn't change the behavior -- having the source code sitting alone under Plugins/ triggers the issue.

> Since I'm using the launcher version and not compiling the editor myself, I'm unable to provide any detailed call stack; or simply don't know how to get one.

### **What I tried**
- Deleting all the generated files (Binaries, Config, Intermediate etc.) -- no change.
- Deleting all .vscode generated files as well -- no change.
- Created a clean C++ project

**Biases**
I'm sort of assuming SUQS worked the same as SUDS, meaning I don't need to compile the editor myself for it to work; as I found nothing in the documentation that suggested otherwise.

Any idea on what could cause this, and how to fix it?
Thanks!!

Thanks a lot for the reply!
I was able to find the culprit in the logs (which I'm not sure why I haven't looked there in the first place):

```
[3/11] Compile [x64] Module.SUQS.3.cpp
D:\Projects\Canopy\Project\Plugins\SUQS\Source\SUQS\Private\SuqsQuestState.cpp(476): error C2440: 'type cast': cannot convert from 'ESuqsObjectiveStatus' to 'ESuqsQuestStatus'
D:\Projects\Canopy\Project\Plugins\SUQS\Source\SUQS\Private\SuqsQuestState.cpp(476): note: Conversion to enumeration type requires an explicit cast (static_cast, C-style cast or parenthesized function-style cast)
D:\Projects\Canopy\Project\Plugins\SUQS\Source\SUQS\Private\SuqsQuestState.cpp(476): error C2046: illegal case
[4/11] Compile [x64] Module.SUQS.2.cpp
[5/11] Compile [x64] Module.SUQS.1.cpp
[6/11] Compile [x64] Module.SUQSTest.cpp
D:\Projects\Canopy\Project\Plugins\SUQS\Source\SUQSTest\Private\TestQuestEvents.cpp(8): warning C4996: PRAGMA_DISABLE_OPTIMIZATION has been deprecated. Use UE_DISABLE_OPTIMIZATION for temporary development or UE_DISABLE_OPTIMIZATION_SHIP to submit Please update your code to the new API before upgrading to the next release, otherwise your project will no longer compile.
D:\Projects\Canopy\Project\Plugins\SUQS\Source\SUQSTest\Private\TestQuestEvents.cpp(618): warning C4996: PRAGMA_ENABLE_OPTIMIZATION has been deprecated. Use UE_ENABLE_OPTIMIZATION for temporary development or UE_ENABLE_OPTIMIZATION_SHIP to submit Please update your code to the new API before upgrading to the next release, otherwise your project will no longer compile.
[7/11] Link [x64] UnrealEditor-SUQSTest.lib
```

The error is about this switch case:

``` cpp
case ESuqsObjectiveStatus::Failed:
	Progression->RaiseQuestFailed(this);
	break;
```

I changed it to 

``` cpp
case ESuqsQuestStatus::Failed:
	Progression->RaiseQuestFailed(this);
	break;
```

Which compiles fine in 5.3.1!

Edit: note that I haven't touched the test files that just throw a warning as I'm not sure those deprecated flags are 5.3+ specific or not.

--- a/Source/SUQS/Private/SuqsQuestState.cpp
+++ b/Source/SUQS/Private/SuqsQuestState.cpp
@@ -473,7 +473,7 @@ void USuqsQuestState::ChangeStatus(ESuqsQuestStatus NewStatus)
 		case ESuqsQuestStatus::Completed: 
 			Progression->RaiseQuestCompleted(this);
 			break;
-		case ESuqsObjectiveStatus::Failed:
+		case (ESuqsQuestStatus)ESuqsObjectiveStatus::Failed:
 			Progression->RaiseQuestFailed(this);
 			break;
 		case ESuqsQuestStatus::Incomplete:

--- a/Source/SUQS/Private/SuqsQuestState.cpp
+++ b/Source/SUQS/Private/SuqsQuestState.cpp
@@ -473,7 +473,7 @@ void USuqsQuestState::ChangeStatus(ESuqsQuestStatus NewStatus)
 		case ESuqsQuestStatus::Completed: 
 			Progression->RaiseQuestCompleted(this);
 			break;
-		case (ESuqsQuestStatus)ESuqsObjectiveStatus::Failed:
+		case ESuqsQuestStatus::Failed:
 			Progression->RaiseQuestFailed(this);
 			break;
 		case ESuqsQuestStatus::Incomplete:

